pimd/pim_cmd_common.c

```diff
@@ -5705,3 +5600,34 @@ int pim_show_bsm_db_helper(const char *vrf, struct vty *vty, bool uj)
 
 	return CMD_SUCCESS;
 }
+
+int pim_router_config_write(struct vty *vty)
+{
+	struct vrf *vrf;
+	struct pim_instance *pim;
+	int writes = 0;
+	char framestr[64] = { 0 };
+
+	RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
+		pim = vrf->info;
+
+		if (!pim)
+			continue;
+
+		snprintfrr(framestr, sizeof(framestr), "router pim%s",
+			   (PIM_AF == AF_INET6 ? "6" : ""));
```

Comment:
Common code between pim and pim6. `PIM_AF == AF_INET6 ` triggers 
```
WARNING: Comparisons should place the constant on the right side of the test
```
The warning is trigged regardless  of the order, a rare use case.